tesseract-server/src/handlers/logic_layer/aggregate.rs

```diff
@@ -206,3 +332,682 @@ pub fn logic_layer_aggregation(
+        ll_config: &Option<LogicLayerConfig>
+) -> Result<HashMap<String, String>, Error> {
+    // Holds a mapping from cut keys (dimensions or levels) to values
+    let mut agg_query_opt_cuts = match &agg_query_opt.cuts {
+        Some(c) => c.clone(),
+        None => HashMap::new()
+    };
+
+    // Process `time` param (latest/oldest)
+    match &agg_query_opt.time {
+        Some(time_param) => {
+            let time_cuts: Vec<String> = time_param.split(",").map(|s| s.to_string()).collect();
+
+            for time_cut in time_cuts {
+                let tc: Vec<String> = time_cut.split(".").map(|s| s.to_string()).collect();
+
+                if tc.len() != 2 {
+                    return Err(format_err!("Malformatted time cut"));
+                }
+
+                let time = match Time::from_key_value(tc[0].clone(), tc[1].clone()) {
+                    Ok(time) => time,
+                    Err(err) => return Err(format_err!("{}", err.to_string()))
+                };
+
+                let (cut, cut_value) = match cube_cache.get_time_cut(time) {
+                    Ok(cut) => cut,
+                    Err(err) => return Err(format_err!("{}", err.to_string()))
+                };
+
+                agg_query_opt_cuts.insert(cut, cut_value);
+            }
+        },
+        None => ()
+    };
+
+    // Find and perform any named set substitutions
+    for (cut_key, cut_values) in agg_query_opt_cuts.clone().iter() {
+        if cut_values.is_empty() {
+            continue;
+        }
+
+        let mut final_cuts: Vec<String> = vec![];
+
+        let cut_values_split: Vec<String> = cut_values.split(",").map(|s| s.to_string()).collect();
+
+        for cut_value in &cut_values_split {
+            match ll_config.clone() {
+                Some(ll_conf) => {
+                    let new_cut_values = ll_conf.substitute_cut(cut_key.clone(), cut_value.clone());
+
+                    if &new_cut_values != cut_value {
+                        let new_cut_values_split: Vec<String> = new_cut_values.split(",").map(|s| s.to_string()).collect();
+
+                        final_cuts = [&final_cuts[..], &new_cut_values_split[..]].concat();
+                    } else {
+                        final_cuts.push(new_cut_values.clone());
+                    }
+                },
+                None => {
+                    final_cuts.push(cut_value.clone());
+                }
+            };
+        }
+
+        *agg_query_opt_cuts.get_mut(cut_key).unwrap() = final_cuts.join(",");
+    }
+
+    Ok(agg_query_opt_cuts)
+}
+
+
+/// Implements logic to resolve logic layer cuts (including those with operations)
+/// into a HashMap separating cuts for each dimension. Doing so helps generate all
+/// the possible cut combinations in the next step.
+/// This method also returns a HashMap of header name substitutions that will help
+/// with the naming of the final column names in the response.
+pub fn resolve_cuts(
```

Comment:
Also just checking if there's any big changes here since it's a big diff.

Reply:
I moved the core of the cut resolution logic out of the `generate_ts_queries` function into this one because it was getting too big. So here is where I am doing a lot of the logic for dealing with cut operations.